src/main/java/hudson/tasks/junit/pipeline/JUnitResultsStepExecution.java

```diff
@@ -59,22 +61,45 @@ protected TestResultSummary run() throws Exception {
     /**
      * Get the stage and parallel branch start node IDs (not the body nodes) for this node, innermost first.
      * @param node A flownode.
-     * @return A nonnull, possibly empty list of stage/parallel branch start node IDs, innermost first.
+     * @return A nonnull, possibly empty list of stage/parallel branch start nodes, innermost first.
      */
     @Nonnull
-    public static List<String> getEnclosingStagesAndParallels(FlowNode node) {
-        List<String> enclosingBlocks = new ArrayList<>();
+    public static List<FlowNode> getEnclosingStagesAndParallels(FlowNode node) {
+        List<FlowNode> enclosingBlocks = new ArrayList<>();
         for (FlowNode enclosing : node.getEnclosingBlocks()) {
             if (enclosing != null && enclosing.getAction(LabelAction.class) != null) {
                 if ((enclosing instanceof StepStartNode && ((StepStartNode) enclosing).getDescriptor() instanceof StageStep.DescriptorImpl) ||
                         (enclosing.getAction(ThreadNameAction.class) != null)) {
-                    enclosingBlocks.add(enclosing.getId());
+                    enclosingBlocks.add(enclosing);
```

Comment:
🐛  Why aren't we getting the ID?

Reply:
Because I decided to change this to return a list of `FlowNode`s instead - pretty much every time I was calling it, I wanted the `FlowNode`s rather than just the IDs, so hey.

Reply:
D'oh, misread that.  Okay, approving!